pkg/reconciler/ingress/ingress.go

```diff
@@ -213,6 +229,8 @@ func (r *Reconciler) reconcileIngress(ctx context.Context, ing *v1alpha1.Ingress
 		ing.Status.MarkLoadBalancerNotReady()
 	}
 
+	ing.Status.ObservedGeneration = ing.GetGeneration()
```

Comment:
Maybe document why this needs to be here.
Also, as suggested it might be safer to copy the ingress object so it is crystal clear when we mutate the new one and query the old one.

Reply:
I think I prefer this approach; it seems unnecessary to perform an extra copy to get the same result. Plus it'd add confusion as to which version we need to be processing on.

IMO the ObservedGeneration update should always have been here, as we weren't following the documentation for ObservedGeneration's purpose:
```
	// ObservedGeneration is the 'Generation' of the Service that
	// was last processed by the controller.
	// +optional
```
Before, we were changing it before processing. So if anything happened during processing we would've updated it incorrectly. I'll add a comment though